node_modules/oae-zencoder/tests/test-zencoder-api.js

```diff
@@ -0,0 +1,196 @@
+/*
+ * Copyright 2016 Apereo Foundation (AF) Licensed under the
+ * Educational Community License, Version 2.0 (the 'License'); you may
+ * not use this file except in compliance with the License. You may
+ * obtain a copy of the License at
+ *
+ *     http://opensource.org/licenses/ECL-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an 'AS IS'
+ * BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+var _ = require('underscore');
+var assert = require('assert');
+var fs = require('fs');
+var util = require('util');
+
+var RestAPI = require('oae-rest');
+var TestsUtil = require('oae-tests/lib/util');
+
+var ZencoderAPI = require('oae-zencoder/lib/api');
+var ZencoderDAO = require('oae-zencoder/lib/internal/dao');
+
+describe('Zencoder API', function() {
+
+    var globalAdminRestContext = null;
+    var camAdminRestContext = null;
+    var globalAdminOnTenantRestContext = null;
+
+    /*!
+     * Set up rest contexts before each test
+     */
+    beforeEach(function(callback) {
+        globalAdminRestContext = TestsUtil.createGlobalAdminRestContext();
+        camAdminRestContext = TestsUtil.createTenantAdminRestContext(global.oaeTests.tenants.cam.host);
+
+        // Log in the admin so his cookie jar is set up appropriately
+        RestAPI.Admin.loginOnTenant(globalAdminRestContext, 'localhost', null, function(err, ctx) {
+            assert.ok(!err);
+            globalAdminOnTenantRestContext = ctx;
+
+            RestAPI.User.getMe(globalAdminOnTenantRestContext, function(err, user) {
+                assert.ok(!err);
+                assert.ok(!user.anon);
+                callback();
+            });
+        });
+    });
+
+    /*!
+     * @return a stream to a video file
+     */
+    var _getVideoStream = function() {
+        return fs.createReadStream(__dirname + '/data/video.mp4');
+    };
+
+    describe('#encodingCompleted', function() {
+        it('verify a valid notification is processed correctly', function(callback) {
+            TestsUtil.generateTestUsers(globalAdminOnTenantRestContext, 1, function(err, users) {
+                assert.ok(!err);
+                var user = _.values(users)[0];
+                var restCtx = user.restContext;
+                // Create the content item that we'll test with
+                RestAPI.Content.createFile(restCtx, 'test video', null, 'private', _getVideoStream, null, null, null, function(err, content) {
+                    assert.ok(!err);
+                    ZencoderDAO.saveContentRevisionId('10', content.id, content.latestRevisionId, function(err) {
```

Comment:
Here is an example of some of our preview tests: https://github.com/oaeproject/Hilary/blob/master/node_modules/oae-preview-processor/tests/test-previews.js#L356

You can see how we bail on them if previews aren't enabled in the config.js. There are utilities shown in there that wait for messages to make it through rabbitmq etc...

I think the easiest way to test may be to use nock to mock the expected HTTP requests to zencoder while making use of the full preview processing lifecycle, and use the `RestAPI` to stage our callbacks.

Additionally, after the callbacks, it will be important to verify the preview status and data on the revisions to ensure it persists properly, I think those kinds of assertions are currently be overlooked here as well.